### ISSUE TITLE
spawn queue instead of kernel.register?

### ISSUE THREAD

**Issue by JestDotty** [JestDotty/screep.serfinity#85]
might help with global.k assignment..

may or may not be possible.... would be cleaner tho

**Comment by JestDotty** [comment]
seems would lose ability to use spawned items while spawning them...

maybe there's another way of making it cleaner while keeping the registration model, but otherwise this results in a bunch of bugs + is more limiting so i'm scrapping it

**Comment by JestDotty** [comment]
fix defense not being sorted into the right run order (spawned by hive, was never running first tick, now runs after parent's spawn if should've run before parent's spawn):

```
[19:17:53]DEBUG  PSR 2fd4a125-d0c4-4c84-8fe8-956d682acff5:2:10848	 hive/wax 3 ["driver/genesis[1|1]","driver/collective[2|2]","hive[4|9]","hive/wax[5|20]","driver/broker[3|30]"]
[19:17:53]DEBUG  PSR 2fd4a125-d0c4-4c84-8fe8-956d682acff5:2:10848	 hive/defense 3 ["driver/genesis[1|1]","driver/collective[2|2]","hive[4|9]","hive/defense[6|5]","hive/wax[5|20]","driver/broker[3|30]"]
```

**Comment by JestDotty** [comment]
![image](https://user-images.githubusercontent.com/4729968/32984075-048a4c48-cc6d-11e7-878a-fd59f25f9b29.png)


```
driver/collective[1|1|A]
driver/collective[1|1|A]
driver/genesis[2|2|A]
driver/genesis[2|2|A]
hive/defence[6|5|A]
driver/broker[3|30|A]
hive[4|9|A]
driver/broker[3|30|A]
hive/wax[5|20|A]
hive/defence[6|5|A]
hive[4|9|A]
hive/dynamo[7|10|A]
hive/dynamo[8|10|A]
hive/dynamo[9|10|A]
hive/dynamo[10|10|A]
hive/architect[11|10|A]
hive/temple[12|10|A]
hive/logistics[13|10|A]
hive/wax[5|20|A]
job/tonic[14|10|A]
job/tonic[15|10|A]
job/tonic[16|10|A]
hive/dynamo[7|10|A]
job/tonic[17|10|A]
job/tonic[18|10|A]
job/tonic[19|10|A]
hive/dynamo[8|10|A]
job/tonic[20|10|A]
job/tonic[21|10|A]
job/tonic[22|10|A]
job/tonic[23|10|A]
hive/dynamo[9|10|A]
job/tonic[24|10|A]
hive/dynamo[10|10|A]
job/veneer[25|10|A]
job/veneer[26|10|A]
job/veneer[27|10|A]
job/veneer[28|10|A]
job/veneer[29|10|A]
job/veneer[30|10|A]
job/veneer[31|10|A]
job/veneer[32|10|A]
job/veneer[33|10|A]
job/veneer[34|10|A]
hive/architect[11|10|A]
hive/temple[12|10|A]
job/mule[35|10|A]
job/mule[36|10|A]
job/mule[37|10|A]
job/mule[38|10|A]
job/mule[39|10|A]
job/mule[40|10|A]
job/mule[41|10|A]
job/mule[42|10|A]
job/mule[43|10|A]
job/mule[44|10|A]
job/mule[45|10|A]
job/mule[46|10|A]
job/mule[47|10|A]
job/mule[48|10|A]
job/mule[49|10|A]
job/mule[50|10|A]
job/mule[51|10|A]
job/mule[52|10|A]
job/mule[53|10|A]
job/mule[54|10|A]
job/mule[55|10|A]
hive/logistics[13|10|A]
job/tonic[14|10|A]
job/tonic[15|10|A]
job/tonic[16|10|A]
job/tonic[17|10|A]
job/tonic[18|10|A]
job/tonic[19|10|A]
job/tonic[20|10|A]
job/tonic[21|10|A]
job/tonic[22|10|A]
job/tonic[23|10|A]
job/tonic[24|10|A]
job/veneer[25|10|A]
job/veneer[26|10|A]
job/veneer[27|10|A]
job/veneer[28|10|A]
job/veneer[29|10|A]
job/veneer[30|10|A]
job/veneer[31|10|A]
job/veneer[32|10|A]
job/veneer[33|10|A]
job/veneer[34|10|A]
job/mule[35|10|A]
job/mule[36|10|A]
job/mule[37|10|A]
job/mule[38|10|A]
job/mule[39|10|A]
job/mule[40|10|A]
job/mule[41|10|A]
job/mule[42|10|A]
job/mule[43|10|A]
job/mule[44|10|A]
job/mule[45|10|A]
job/mule[46|10|A]
job/mule[47|10|A]
job/mule[48|10|A]
job/mule[49|10|A]
job/mule[50|10|A]
job/mule[51|10|A]
job/mule[52|10|A]
job/mule[53|10|A]
job/mule[54|10|A]
job/mule[55|10|A]
```

**Comment by JestDotty** [comment]
fix wax not spawning at the wrong time:

![image](https://user-images.githubusercontent.com/4729968/32985862-1e724a78-cc92-11e7-9c7b-97d3d5881309.png)

**Comment by JestDotty** [comment]
looks good so far... sim and pserver with RCL 6

**Comment by JestDotty** [comment]
```
Hive[4]-sim {"name":"hive","ppid":2,"args":["sim"],"order":9,"pid":4,"m":{"inventory":{"store":{"capacity":0},"carry":{"capacity":250,"energy":250},"drops":{"energy":5944},"operation":{"capacity":300,"energy":300}},"name":"sim","structures":{"91bf9c21a1ec7f2b173e6afc":{},"79f0d4a477c7e1c89b6f45ac":{}},"constructions":{"b36c1ce2b59d12f3be24bfb4":{"x":33,"y":22},"21b24035b0e282b82bcfb9f1":{"x":33,"y":3},"1ea7e815ef384d27b9f5110c":{"x":41,"y":42},"80bd9458b884ae71d827c578":{"x":40,"y":41},"46c57b386507233acfa3decb":{"x":39,"y":40},"f4d8b28303127df6b1c6c6c3":{"x":38,"y":39},"8251034566d12e920b664524":{"x":5,"y":42}},"drops":{"b6f2c8dd9080ddf8e1b2c171":{},"bcfec458673bb11382b7461b":{},"c60457da59136a4392787bc3":{},"62dee9c1b5bb1929a53887bb":{},"f5ab65de05b1b61d2ea17be0":{},"d50edfe79b4aa4e092ba2c79":{},"51be1e1ce58978a135d23a32":{},"e5ab4e49344be8a23b3c2e71":{}},"hitsMax":100000,"emergencyRepair":0.2,"danger":false,"wax":5,"defence":6,"dynamo":{"e5f640bda39baf5b8e4af608":7,"15b5b5d0f8551c9a84cde60b":8,"f1c9cf900551b9ad1216d4f3":9,"a742fa8359fb98c804d1bdcd":10},"architect":11,"upgrading":12,"logistics":13},"errs":1} 
 TypeError: Cannot read property 'count' of undefined
    at m.emergencySpawn._.sum.pid (eval at exports.evalCode (blob:https://screeps.com/2fd4a125-d0c4-4c84-8fe8-956d682acff5:2:10848), :58:82)
    at nt (blob:https://screeps.com/2fd4a125-d0c4-4c84-8fe8-956d682acff5:2:27480)
    at Function.A.t.sum (blob:https://screeps.com/2fd4a125-d0c4-4c84-8fe8-956d682acff5:2:63975)
    at Hive.state (eval at exports.evalCode (blob:https://screeps.com/2fd4a125-d0c4-4c84-8fe8-956d682acff5:2:10848), :58:29)
    at Hive.exec (eval at exports.evalCode (blob:https://screeps.com/2fd4a125-d0c4-4c84-8fe8-956d682acff5:2:10848), :54:8)
    at process (eval at exports.evalCode (blob:https://screeps.com/2fd4a125-d0c4-4c84-8fe8-956d682acff5:2:10848), :85:17)
    at Processor.process (eval at exports.evalCode (blob:https://screeps.com/2fd4a125-d0c4-4c84-8fe8-956d682acff5:2:10848), :96:11)
    at data.cpu.profiler.profile (eval at exports.evalCode (blob:https://screeps.com/2fd4a125-d0c4-4c84-8fe8-956d682acff5:2:10848), :84:12)
    at Object.profile (eval at exports.evalCode (blob:https://screeps.com/2fd4a125-d0c4-4c84-8fe8-956d682acff5:2:10848), :4:3)
    at Processor.exec (eval at exports.evalCode (blob:https://screeps.com/2fd4a125-d0c4-4c84-8fe8-956d682acff5:2:10848), :82:26)
```

oops, forgot about counts not working... that's not as intended...

**Comment by JestDotty** [comment]
fix count by changing to sprout queue instead of having 1 running queue

also made interrupter a bit more complicated but fixed some bug in it #90

```
0 driver/collective[1|A|1]
1 driver/genesis[2|A|2]
2 hive[4|A|9]
3 hive/defence[6|A|5]
4 hive/dynamo[7|A|10]
5 hive/dynamo[8|A|10]
6 hive/dynamo[9|A|10]
7 hive/dynamo[10|A|10]
8 hive/architect[11|A|10]
9 hive/temple[12|A|10]
10 hive/logistics[13|A|10]
11 job/tonic[14|A|10]
12 hive/wax[5|A|20]
13 driver/broker[3|A|30]
14 job/tonic[15|A|10]
15 job/tonic[16|A|10]
16 job/tonic[17|A|10]
17 job/tonic[18|A|10]
18 job/tonic[19|A|10]
19 job/tonic[20|A|10]
20 job/tonic[21|A|10]
21 job/tonic[22|A|10]
22 job/tonic[23|A|10]
23 job/tonic[24|A|10]
24 job/veneer[25|A|10]
25 job/veneer[26|A|10]
26 job/veneer[27|A|10]
27 job/veneer[28|A|10]
28 job/veneer[29|A|10]
29 job/veneer[30|A|10]
30 job/veneer[31|A|10]
31 job/veneer[32|A|10]
32 job/veneer[33|A|10]
33 job/veneer[34|A|10]
34 job/mule[35|A|10]
35 job/mule[36|A|10]
36 job/mule[37|A|10]
37 job/mule[38|A|10]
38 job/mule[39|A|10]
39 job/mule[40|A|10]
40 job/mule[41|A|10]
41 job/mule[42|A|10]
42 job/mule[43|A|10]
43 job/mule[44|A|10]
44 job/mule[45|A|10]
45 job/mule[46|A|10]
46 job/mule[47|A|10]
47 job/mule[48|A|10]
48 job/mule[49|A|10]
49 job/mule[50|A|10]
50 job/mule[51|A|10]
51 job/mule[52|A|10]
52 job/mule[53|A|10]
53 job/mule[54|A|10]
54 job/mule[55|A|10] 
ran: 
0 driver/collective[1|A|1]
1 driver/genesis[2|A|2]
2 hive/defence[6|A|5]
3 hive[4|A|9]
4 hive/dynamo[7|A|10]
5 hive/dynamo[8|A|10]
6 hive/dynamo[9|A|10]
7 hive/dynamo[10|A|10]
8 hive/architect[11|A|10]
9 hive/temple[12|A|10]
10 hive/logistics[13|A|10]
11 job/tonic[14|A|10]
12 job/tonic[15|A|10]
13 job/tonic[16|A|10]
14 job/tonic[17|A|10]
15 job/tonic[18|A|10]
16 job/tonic[19|A|10]
17 job/tonic[20|A|10]
18 job/tonic[21|A|10]
19 job/tonic[22|A|10]
20 job/tonic[23|A|10]
21 job/tonic[24|A|10]
22 job/veneer[25|A|10]
23 job/veneer[26|A|10]
24 job/veneer[27|A|10]
25 job/veneer[28|A|10]
26 job/veneer[29|A|10]
27 job/veneer[30|A|10]
28 job/veneer[31|A|10]
29 job/veneer[32|A|10]
30 job/veneer[33|A|10]
31 job/veneer[34|A|10]
32 job/mule[35|A|10]
33 job/mule[36|A|10]
34 job/mule[37|A|10]
35 job/mule[38|A|10]
36 job/mule[39|A|10]
37 job/mule[40|A|10]
38 job/mule[41|A|10]
39 job/mule[42|A|10]
40 job/mule[43|A|10]
41 job/mule[44|A|10]
42 job/mule[45|A|10]
43 job/mule[46|A|10]
44 job/mule[47|A|10]
45 job/mule[48|A|10]
46 job/mule[49|A|10]
47 job/mule[50|A|10]
48 job/mule[51|A|10]
49 job/mule[52|A|10]
50 job/mule[53|A|10]
51 job/mule[54|A|10]
52 job/mule[55|A|10]
53 hive/wax[5|A|20]
54 driver/broker[3|A|30]
```
